packages/flutter/lib/src/material/input_border.dart

```diff
@@ -416,28 +416,30 @@ class OutlineInputBorder extends InputBorder {
       scaledRRect.blRadiusX * 2.0,
     );
 
+    // This assumes that the radius is circular (x and y radius are equal).
+    // Currently, BorderRadius only supports circular radii.
     const double cornerArcSweep = math.pi / 2.0;
-    final double tlCornerArcSweep = start < scaledRRect.tlRadiusX
-      ? math.asin((start / scaledRRect.tlRadiusX).clamp(-1.0, 1.0))
-      : math.pi / 2.0;
+    final double tlCornerArcSweep = math.acos(
+      (1 - start / scaledRRect.tlRadiusX).clamp(0.0, 1.0),
```

Comment:
nit: is it guaranteed that the radiusX can't be 0?

Reply:
TIL! So `NaN.clamp(0, 1) == 1`? But shouldn't the `1 - NaN` be evaluated first and throw an error?

Reply:
turns out, `1 - NaN = -NaN`. So it finally clamps to 0.0.

Reply:
And `1/0` is actually `Infinity` in Dart and not `NaN`, so it makes a little more sense.